pkg/backend/openshiftcluster.go

```diff
@@ -33,6 +34,16 @@ func (ocb *openShiftClusterBackend) try(ctx context.Context) (bool, error) {
 		return false, err
 	}
 
+	r, err := azure.ParseResourceID(doc.OpenShiftCluster.ID)
+	if err != nil {
+		return false, err
+	}
+
+	subscriptionDoc, err := ocb.db.Subscriptions.Get(ctx, r.SubscriptionID)
+	if err != nil || subscriptionDoc == nil {
+		return false, err
```

Comment:
subscriptionDoc should never be nil, but if it actually happened we would incorrectly return `false, nil`.  Separate this case and return a synthetic error if it happens.

Reply:
wouldn't it give the same end result? if subscription doc is nil we are in a bigger trouble. But even if I split this out here - this condition will be checked again on the next cycle. If we return `true, err` then it will be retried immediately. But in either case we are potentially stuck in the loop here.

Reply:
You've got a point.  This code is in the wrong place (too early).  It "should never happen", of course, but if it did, you would get a loop because this code is before the `doc.Dequeues > maxDequeueCount` check, which is definitely wrong.

It is also before the new goroutine + defer recover.Panic + ocb.heartbeat, all of which is also wrong.  

Maybe pass ocb.db into openshiftcluster.NewManager instead of ocb.db.OpenShiftClusters, and do the document fetch inside NewManager.

Reply:
I have moved it just before calling `NewManager`, I think it is cleaner than passing the `db.Subscriptions` because it is not needed for anything else than fetching the subscription doc once, and it will require to pass in `ctx` as well.